### PR TITLE
[Oauth] Developers page

### PR DESCRIPTION
**Pull request by csubira**
https://github.com/CartoDB/product/issues/418
https://github.com/CartoDB/product/issues/424

### CODE REVIEW

File: lib/assets/javascripts/dashboard/data/api-key-model.js [hunk 1]
```diff
@@ -149,7 +149,7 @@ module.exports = Backbone.Model.extend({
   _parseDatabaseSchemas: function (grants) {
     const schemas = _.find(grants, grant => grant.type === GRANT_TYPES.DATABASE).schemas;
     const publicSchema = _.find(schemas, schema => schema.name === 'public');
-    return publicSchema && publicSchema.permissions && publicSchema.permissions.indexOf('create') > -1;
+    return !!(publicSchema && publicSchema.permissions && publicSchema.permissions.indexOf('create') > -1);
```
Comment (jesusbotella) [inline lib/assets/javascripts/dashboard/data/api-key-model.js]: Why do you need to do that? 🤔 
The comparison returns a boolean

File: lib/assets/test/spec/dashboard/views/api-keys/api-keys-form-view.spec.js [hunk 2]
```diff
@@ -375,14 +433,10 @@ describe('dashboard/views/api-keys/api-keys-form-view', function () {
...
-      it('should clean validation tooltip if any', function () {
-        view._validationTooltip = {
-          clean: jasmine.createSpy('clean')
-        };
-
+      it('should set validation tooltip to null if any', function () {
         view._onFormChanged();
 
-        expect(view._validationTooltip.clean).toHaveBeenCalled();
+        expect(view._validationTooltip).toBe(null);
```
Comment (jesusbotella) [inline lib/assets/test/spec/dashboard/views/api-keys/api-keys-form-view.spec.js]: I think we should check that the clean action is performed as well

File: lib/assets/javascripts/dashboard/views/api-keys/api-keys-form-view.js [hunk 3]
```diff
@@ -172,10 +175,31 @@ module.exports = CoreView.extend({
   _onFormChanged: function () {
     this._handleFormErrors();
     this._handleCheckboxState();
-    this._handleCreateCheckbox();
     this.$('.js-error').hide();
   },
 
+  _onClickCreateCheckbox: function () {
+    const createCheckbox = this._formView.getValue().datasets.create;
```
Comment (jesusbotella) [inline lib/assets/javascripts/dashboard/views/api-keys/api-keys-form-view.js]: I don't really get what you're trying to do here. Isn't this change performed by backbone forms?

File: lib/assets/test/spec/dashboard/views/api-keys/api-keys-form-view.spec.js (hunk 2 above)
Comment (csubira) [reply lib/assets/test/spec/dashboard/views/api-keys/api-keys-form-view.spec.js]: 👍

File: lib/assets/javascripts/dashboard/data/api-key-model.js (hunk 1 above)
Comment (csubira) [reply lib/assets/javascripts/dashboard/data/api-key-model.js]: That's what I thought, but when I implemented the tests I saw that it returned `undefined` when there is no `publicSchema` (considering`_.find` returns `undefined` when not found)

File: lib/assets/javascripts/dashboard/views/api-keys/api-keys-form-view.js (hunk 3 above)
Comment (csubira) [reply lib/assets/javascripts/dashboard/views/api-keys/api-keys-form-view.js]: I replaced it for an event on change this checkboxes (as we discussed) and it has to do with forcing SQL checkbox to be checked when CREATE is checked and uncheck CREATE if SQL is unchecked.

**Pull request merged**